File: ios/Breathing/BreathingConfiguration.swift
import SwiftUI

public struct BreathingConfiguration {
    var blobColors: [Color]
    var innerBlobColor: Color
    var glowColor: Color
    var particleColor: Color
    var progressRingColor: Color
    var textColor: Color
    var showProgressRing: Bool
    var showTextCue: Bool
    var showInnerBlob: Bool
    var showShadow: Bool
    var showParticles: Bool
    var showWavyBlobs: Bool
    var showGlowEffects: Bool
    var pointCount: Int
    var wobbleIntensity: Double

    public init(
        blobColors: [Color] = [
            Color(red: 0.4, green: 0.7, blue: 0.9),
            Color(red: 0.3, green: 0.5, blue: 0.8),
            Color(red: 0.5, green: 0.3, blue: 0.7)
        ],
        innerBlobColor: Color = Color.white.opacity(0.3),
        glowColor: Color = .white,
        particleColor: Color = .white,
        progressRingColor: Color = Color.white.opacity(0.5),
        textColor: Color = .white,
        showProgressRing: Bool = true,
        showTextCue: Bool = true,
        showInnerBlob: Bool = true,
        showShadow: Bool = true,
        showParticles: Bool = true,
        showWavyBlobs: Bool = true,
        showGlowEffects: Bool = true,
        pointCount: Int = 8,
        wobbleIntensity: Double = 1.0
    ) {
        self.blobColors = blobColors
        self.innerBlobColor = innerBlobColor
        self.glowColor = glowColor
        self.particleColor = particleColor
        self.progressRingColor = progressRingColor
        self.textColor = textColor
        self.showProgressRing = showProgressRing
        self.showTextCue = showTextCue
        self.showInnerBlob = showInnerBlob
        self.showShadow = showShadow
        self.showParticles = showParticles
        self.showWavyBlobs = showWavyBlobs
        self.showGlowEffects = showGlowEffects
        self.pointCount = pointCount
        self.wobbleIntensity = wobbleIntensity
    }
}

File: android/src/main/java/expo/modules/breathing/BreathingConfiguration.kt
package expo.modules.breathing

import androidx.compose.ui.graphics.Color

data class BreathingConfiguration(
    val blobColors: List<Color> = listOf(
        Color(0xFF66B3E6),  // Light blue
        Color(0xFF4D80CC),  // Medium blue
        Color(0xFF804DB3)   // Purple
    ),
    val innerBlobColor: Color = Color.White.copy(alpha = 0.3f),
    val glowColor: Color = Color.White,
    val particleColor: Color = Color.White,
    val progressRingColor: Color = Color.White.copy(alpha = 0.5f),
    val textColor: Color = Color.White,
    val showProgressRing: Boolean = true,
    val showTextCue: Boolean = true,
    val showInnerBlob: Boolean = true,
    val showShadow: Boolean = true,
    val showParticles: Boolean = true,
    val showWavyBlobs: Boolean = true,
    val showGlowEffects: Boolean = true,
    val pointCount: Int = 8,
    val wobbleIntensity: Double = 1.0
)
